src/components/Calendar/indicators.js

```diff
@@ -1,38 +1,59 @@
-import React, { PropTypes } from 'react';
+import React, { PropTypes, Component } from 'react';
 import { compose } from '../../utils/utils';
 import Day from '../Day/Day';
 import { indicatorDay } from '../Day/indicatorDay';
 import { isSameDay } from '../../utils/date-utils';
 
 /**
- * Higher Order Component to add "day indicators" feature
- * @param {Component} Component
+ * Higher Order Component to add "day indicators" feature to Calendar
+ * @param {Component} CalendarComponent
  * @return {IndicatorCalendar}
  */
-export function indicators(Component) {
-  class IndicatorCalendar extends React.Component {
+export function indicators(CalendarComponent) {
+  class IndicatorCalendar extends Component {
     constructor() {
       super(...arguments);
 
+      this.state = {
+        // Create enhancedDayComponent and store in state.
+        enhancedDayComponent: this.enhanceDayComponent(this.props.DayComponent),
+      };
+    }
+
+    componentDidUpdate(prevProps) {
+      // We don't want to create this on every render, but we also want
+      // to account for the fact that props.DayComponent could change
+      // We only want to re-create if props.DayComponent changed.
+      // Treat this as a "computed property".
+      if (prevProps.DayComponent !== this.props.DayComponent) {
+        const enhancedDayComponent = this.enhanceDayComponent(this.props.DayComponent);
+        this.setState({enhancedDayComponent});
```

Comment:
Why compute and set this on state rather than compute in render and make use of `*ShouldUpdate`?

I feel like this is a more straightforward approach for "deriving/computing" from `props` in many cases since in order to properly go the `*ShouldUpdate` route, I would have to get into checking any other props (in this case `events: Array`) and that seems much more error-prone and case-by-case to me.

To summarize what I mean by this "computed properties as state" approach.

- Initialize `this.state = {fromX: computePropertyFromX(props.x)}`
- Watch for `props.x` changing
- Re-compute `this.setState({fromX: computePropertyFromX(newProps.x)})`